sound/soc/sof/ipc4-topology.c

```diff
@@ -2745,6 +2745,80 @@ static int sof_ipc4_dai_config(struct snd_sof_dev *sdev, struct snd_sof_widget *
 	return 0;
 }
 
+static int sof_ipc4_nhlt_register_dai_links(struct snd_soc_component *scomp)
```

Comment:
adding dai links in the SOF ipc4 code is a major layering violation. This belongs in machine drivers IMHO. The SOF driver should only add CPU dais, not muck with dai links.

Reply:
The problem is topology is loaded after machine driver registers soundcard.  If we want to extract ssp numbers from NHLT, then we have to register the dai links in sof-topology layer...

Reply:
it's a bit too much for me... I am not hot on hard-coding such names in a generic layer. need to think a bit more on all this.